sql/size_utils.sql

```diff
@@ -126,151 +184,280 @@ BEGIN
+        INTO STRICT table_name, schema_name, is_distributed
+        FROM pg_class c
+        INNER JOIN pg_namespace n ON (n.OID = c.relnamespace)
+        INNER JOIN _timescaledb_catalog.hypertable ht ON (ht.schema_name = n.nspname AND ht.table_name = c.relname)
+        WHERE c.OID = main_table;
+
+        CASE WHEN is_distributed THEN
+            RETURN QUERY SELECT * FROM _timescaledb_internal.chunk_remote_size(schema_name, table_name);
+        ELSE
+            RETURN QUERY SELECT *, NULL::NAME FROM _timescaledb_internal.chunk_local_size(schema_name, table_name);
+        END CASE;
+END;
+$BODY$;
+
+-- Get relation size of the chunks of an hypertable
+
+CREATE OR REPLACE FUNCTION chunk_detailed_size_pretty(
+    main_table              REGCLASS
+)
+RETURNS TABLE (chunk_id INT,
+               chunk_table NAME,
+               heap_size  TEXT,
                index_size  TEXT,
                toast_size  TEXT,
-               total_size  TEXT) LANGUAGE PLPGSQL STABLE STRICT
+               total_size  TEXT,
+               node_name NAME 
+               )
+               LANGUAGE PLPGSQL STABLE STRICT
                AS
 $BODY$
 DECLARE
         table_name       NAME;
         schema_name      NAME;
 BEGIN
         RETURN QUERY
-        SELECT pg_size_pretty(table_bytes) as table,
-               pg_size_pretty(index_bytes) as index,
-               pg_size_pretty(toast_bytes) as toast,
-               pg_size_pretty(total_bytes) as total
-               FROM @extschema@.hypertable_relation_size(main_table);
-
+        SELECT hd.chunk_id,
+               hd.chunk_table,
+               pg_size_pretty(heap_bytes) as heap_size,
+               pg_size_pretty(index_bytes) as index_size,
+               pg_size_pretty(toast_bytes) as toast_size,
+               pg_size_pretty(total_bytes) as total_size,
+               hd.data_node
+               FROM @extschema@.chunk_detailed_size(main_table) hd;
 END;
 $BODY$;
 
-
--- Get the per-node relation size of a distributed hypertable across all nodes in a distributed database
--- like pg_relation_size(hypertable)
--- (https://www.postgresql.org/docs/current/functions-admin.html#FUNCTIONS-ADMIN-DBSIZE)
+-- Get sizes of indexes on a hypertable
 --
--- main_table - hypertable to get size of
+-- main_table - hypertable to get index sizes of
 --
 -- Returns:
--- node_name          - Data node hosting part of the table
--- num_chunks         - Number of chunks hosted on the data node
--- table_size         - Pretty output of table_bytes for the data node
--- index_bytes        - Pretty output of index_bytes for the data node
--- toast_bytes        - Pretty output of toast_bytes for the data node
--- total_size         - Pretty output of total_bytes for the data node
-
-CREATE OR REPLACE FUNCTION hypertable_data_node_relation_size(
+-- index_name           - index on hyper table
+-- total_bytes          - size of index on disk
+
+CREATE OR REPLACE FUNCTION indexes_relation_size(
```

Comment:
We already have the aggregated index information in hypertable_detailed_size and chunks_detailed_size. The only reason I have retained it is because it breaks it down per index.

Reply:
* Still see "relation" in the name. 
* Why is the index name returned as a TEXT field and not schema+name? This is yet a different way of returning this information and it is not consistent with neither `chunks_detailed_size` nor `hypertable_detailed_size`.

Reply:
This PR modifies only adds chunk_detailed_size and hypertable_detailed_size information. and needed dependencies.  The indexes function came up as part of the PR review.The changes being suggested are more extensive than simply removing a function.  So I am going to deal with changes to indexes_relation_size in a follow-up PR.